File: sdks/python/examples/agent_with_pinata_ipfs.py
import asyncio
from os import getenv

from alith.data.storage import (
    GetShareLinkOptions,
    PinataIPFS,
    StorageError,
    UploadOptions,
)


async def main():
    ipfs = PinataIPFS()
    try:
        data = "Your data"
        name = "your_file.txt"
        token = getenv("IPFS_JWT", "")

        file_meta = await ipfs.upload(
            UploadOptions(name=name, data=data.encode(), token=token)
        )
        print(f"Upload file to the Pinata IPFS: {file_meta}")
        print(
            f"Share link: {await ipfs.get_share_link(GetShareLinkOptions(token=token, id=file_meta.id))}"
        )
    except StorageError as e:
        print(f"Error: {e}")
    finally:
        await ipfs.close()


if __name__ == "__main__":
    asyncio.run(main())

File: sdks/node/examples/agent_with_pinata_ipfs.ts
import { PinataIPFS } from "alith/data/storage";

async function main() {
  const ipfs = new PinataIPFS();
  const data = "Your privacy data";
  const name = "your_privacy_file.txt";
  const fileMeta = await ipfs.upload({
    name: name,
    data: Buffer.from(data, "utf-8"),
    token: process.env.IPFS_JWT || "",
  });
  console.log(`Upload file to the Pinata IPFS: ${fileMeta}`);
  console.log(
    `Share link: ${await ipfs.getShareLink({
      token: process.env.IPFS_JWT || "",
      id: fileMeta.id,
    })}`
  );
}

await main();
